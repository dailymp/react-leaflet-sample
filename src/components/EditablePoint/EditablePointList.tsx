import * as React from "react";
import { Point } from "../../common/Point";
import { EditablePoint } from "./EditablePoint";

interface PropsEditablePointList {
    pins: Point[];
    onEditPins: (newPins: Point[]) => void;
    setShowPointList: React.Dispatch<React.SetStateAction<boolean>>;
}

export interface PointWithEnable extends Point {
    isEnable: boolean;
}

export const marginComponentsInEditablePointList = () => ({marginLeft: '1em'});

const parseGetAndSetPinListToPins = (getAndSetPin: [PointWithEnable, React.Dispatch<React.SetStateAction<PointWithEnable>>][]): Point[] => (
    getAndSetPin.map(([pin, _]) => ({
        x: pin.x,
        y: pin.y,
        hexadecimalColor: pin.hexadecimalColor,
    }))
);

export const EditablePointList = (props: PropsEditablePointList) => {
    const getAndSetPin = props.pins.map((pin: Point) => {
        return React.useState({...pin, isEnable: true});
    });
    let indexPin = 0;

    return (
        <>
            <p>
                <button
                    onClick={evt => {
                        props.onEditPins(parseGetAndSetPinListToPins(getAndSetPin));
                        props.setShowPointList(false);
                    }}
                >Apply Changes</button>
                <button
                    style={marginComponentsInEditablePointList()}
                    onClick={evt => props.setShowPointList(false)}
                >Cancel Changes</button>
            </p>
            {
                getAndSetPin.map(([pin, setPin]) => {
                    const result = pin.isEnable ?
                        <EditablePoint
                            key={`editable-point-${indexPin}`}
                            pin={pin}
                            setPin={setPin} />
                        : undefined;
                    indexPin++;
                    return result;
                })
            }
        </>
    );
}

