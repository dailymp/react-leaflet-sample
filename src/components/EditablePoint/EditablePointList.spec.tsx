import * as React from 'react';
import { shallow } from 'enzyme';
import { EditablePointList } from './EditablePointList';

describe('Tests for EditablePointList component', () => {
  it('should render as expected when passing required properties', () => {
    const props = {
        pins: [
            {x: 1, y:3, hexadecimalColor: '#00e700'},
            {x: 4, y:6, hexadecimalColor: '#00e700'},
        ],
        onEditPins: jest.fn(),
        setShowPointList: jest.fn()
    };

    const component = shallow(<EditablePointList
        pins={props.pins}
        onEditPins={props.onEditPins}
        setShowPointList={props.setShowPointList}
    />);

    expect(component).toMatchSnapshot();
  });
});