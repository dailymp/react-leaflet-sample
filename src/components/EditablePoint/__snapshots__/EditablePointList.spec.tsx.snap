// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for EditablePointList component should render as expected when passing required properties 1`] = `
<Fragment>
  <p>
    <button
      onClick={[Function]}
    >
      Apply Changes
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    >
      Cancel Changes
    </button>
  </p>
  <Component
    key="editable-point-0"
    pin={
      Object {
        "hexadecimalColor": "#00e700",
        "isEnable": true,
        "x": 1,
        "y": 3,
      }
    }
    setPin={[Function]}
  />
  <Component
    key="editable-point-1"
    pin={
      Object {
        "hexadecimalColor": "#00e700",
        "isEnable": true,
        "x": 4,
        "y": 6,
      }
    }
    setPin={[Function]}
  />
</Fragment>
`;
